discovery/discoverer.go

```diff
@@ -0,0 +1,7 @@
+package discovery
```

Comment:
Good question!

We moved `samplers` into a separate package when we created the S3 plugin. It made sense for the s3 plugin to be a separate package, but it needed to be able to import the `DDMetric`, and Go doesn't allow cyclical imports, so that meant that `DDMetric` had to be somewhere that both `veneur` and `plugins/s3` could import. In theory, we could have just put `DDMetric` there and not worried about the `Count`/`Histo`/etc. type, but that seems weird.

More broadly: Go's a bit less package-happy than most other languages. Not to an extreme, but it's more common to split things into multiple packages when it becomes necessary (as happened here), as opposed to pre-emptively grouping things logically. That's not an explicit design idiom; just an observation I've found.